moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp

```diff
@@ -214,90 +226,99 @@ bool planning_scene_monitor::CurrentStateMonitor::haveCompleteState(std::vector<
       continue;
-    std::map<const moveit::core::JointModel*, ros::Time>::const_iterator it = joint_time_.find(joint);
+    std::map<const moveit::core::JointModel*, rclcpp::Time>::const_iterator it = joint_time_.find(joint);
     if (it == joint_time_.end())
     {
-      ROS_DEBUG("Joint '%s' has never been updated", joint->getName().c_str());
+      RCLCPP_DEBUG(logger,"Joint '%s' has never been updated", joint->getName().c_str());
       result = false;
     }
     else if (it->second < old)
     {
-      ROS_DEBUG("Joint '%s' was last updated %0.3lf seconds ago (older than the allowed %0.3lf seconds)",
-                joint->getName().c_str(), (now - it->second).toSec(), age.toSec());
+      RCLCPP_DEBUG(logger,"Joint '%s' was last updated %0.3lf seconds ago (older than the allowed %0.3lf seconds)",
+                joint->getName().c_str(), (now - it->second) , age.seconds());
       result = false;
     }
   }
   return result;
 }
 
-bool planning_scene_monitor::CurrentStateMonitor::haveCompleteState(const ros::Duration& age,
+bool planning_scene_monitor::CurrentStateMonitor::haveCompleteState(const rclcpp::Duration& age,
                                                                     std::vector<std::string>& missing_states) const
 {
   bool result = true;
   const std::vector<const moveit::core::JointModel*>& joints = robot_model_->getActiveJointModels();
-  ros::Time now = ros::Time::now();
-  ros::Time old = now - age;
+  rclcpp::Clock ros_clock;
+  rclcpp::Time now = ros_clock.now();
+  rclcpp::Time old = now - age;
   boost::mutex::scoped_lock slock(state_update_lock_);
   for (const moveit::core::JointModel* joint : joints)
   {
     if (joint->isPassive() || joint->getMimic())
       continue;
-    std::map<const moveit::core::JointModel*, ros::Time>::const_iterator it = joint_time_.find(joint);
+    std::map<const moveit::core::JointModel*, rclcpp::Time>::const_iterator it = joint_time_.find(joint);
     if (it == joint_time_.end())
     {
-      ROS_DEBUG("Joint '%s' has never been updated", joint->getName().c_str());
+      RCLCPP_DEBUG(logger,"Joint '%s' has never been updated", joint->getName().c_str());
       missing_states.push_back(joint->getName());
       result = false;
     }
     else if (it->second < old)
     {
-      ROS_DEBUG("Joint '%s' was last updated %0.3lf seconds ago (older than the allowed %0.3lf seconds)",
-                joint->getName().c_str(), (now - it->second).toSec(), age.toSec());
+      RCLCPP_DEBUG(logger,"Joint '%s' was last updated %0.3lf seconds ago (older than the allowed %0.3lf seconds)",
+                joint->getName().c_str(), (now - it->second) , age.seconds());
       missing_states.push_back(joint->getName());
       result = false;
     }
   }
   return result;
 }
 
-bool planning_scene_monitor::CurrentStateMonitor::waitForCurrentState(const ros::Time t, double wait_time) const
-{
-  ros::WallTime start = ros::WallTime::now();
```

Comment:
@wjwwood As you mentioned on https://github.com/ros-planning/moveit2/issues/31 the walltime calls should be replaced with ``std::chrono::system_clock``, so I guess that the correct use of WallDuration should be something like ``std::chrono::duration`` or is there a better way to do it, like just using ``rclcpp::Duration`` calls?

Reply:
`std::chrono::duration` is sufficient I think. We were supposed to have to/from functions for converting between `std::chrono::duration` and `rclcpp::Duration`, but I don't know if they got implemented or not. It would be a good thing to contribute if they're missing.

I don't know what the context here is, so I can't advise you. Please give me an exact question or scenario and I can help more.

Reply:
Basically the question was: What is the best replacemanet for ros::WallDuration/ros::WallTime since they are not implemented for ros2. You have already answered me with, use _std::chrono::system_clock_ as replacement for _ros::walltime_ and _std::chrono::duration_ for _ros::WallDuration_, right?